### PR TITLE
Remove outdated access-data-style fields on ir.model.fields

### PR DESCRIPTION
**Pull request by xmo-odoo**
@moylop260's odoo/odoo#50516 led me to take a look, notice that create_date and write_date were pretty much never set on ir.model.data records and the it also had `date_init`/`date_update` which are not reliably set, and seem like redundant ad-hoc versions of `create_date` and `write_date`.

odoo/odoo#34988 removed them constrains & relations, but we might as well also nuke them from ir.model.data, and fix the various points where a date should be set and is not:

* removes the date_init/date_update fields from the SQL and model
* converts the one bit of code which did set those to set create_date/write_date
* use create_date/write_date in the view
* adds setting those columns to a bunch of raw SQL queries which were missing them (also updates the queries some to merge the literal values into the queries as it seems unnecessary to interpolate e.g. a boolean literal)

Assuming this is considered a good idea, will require some migrationy bits:

* copy date_init and date_update over to `create_date` and `write_date`
* possibly update the schema to set default values on create_date and write_date if we decide to do that (?)

### CODE REVIEW

**Comment by xmo-odoo** [comment]
@mart-e what do you think about putting an SQL-level default on create_date/write_date? This would be somewhat alien as other uses of these fields don't have them, but would ensure all xids at least have those values set even if we forget to specify them explicitly (especially as xids are *generally* interacted with using raw SQL rather than using the ORM).

**Comment by xmo-odoo** [comment]
> @mart-e what do you think about putting an SQL-level default on create_date/write_date? This would be somewhat alien as other uses of these fields don't have them, but would ensure all xids at least have those values set even if we forget to specify them explicitly (especially as xids are _generally_ interacted with using raw SQL rather than using the ORM).
> 
> Maybe a custom UPDATE trigger for write_date as well?

**Comment by mart-e** [comment]
@xmo-odoo maybe SQL experts like @nseinlet will have arguments against that but it SQL default could make sense. Not sure the performance impact of a trigger but a bit odd to reimplement the job of the ORM in SQL

**Comment by nseinlet** [comment]
SQL level default value is great.

File: odoo/modules/db.py [hunk 1]
```diff
@@ -62,9 +62,10 @@ def initialize(cr):
             info['application'], info['icon'],
             info['sequence'], info['summary']))
         id = cr.fetchone()[0]
-        cr.execute('INSERT INTO ir_model_data \
-            (name,model,module, res_id, noupdate) VALUES (%s,%s,%s,%s,%s)', (
-                'module_'+i, 'ir.module.module', 'base', id, True))
+        cr.execute("""
+            INSERT INTO ir_model_data (name,model,module, res_id, noupdate, create_date, write_date)
+            VALUES (%s,'ir.module.module','base',%s,true,now() at time zone 'UTC', now() at time zone 'UTC')
+            """, ('module_'+i, id))
```
Comment (rco-odoo) [inline odoo/modules/db.py:68]: I am in favor of using SQL defaults for `create_date` and `write_date`, as this enables to simplify code in several places, without introducing too much magic.
But what about `create_uid` and `write_uid`?  Those ones are apparently *never* assigned.

File: odoo/addons/base/data/base_data.sql [hunk 2]
```diff
@@ -127,21 +126,26 @@ CREATE TABLE res_partner (
...
 insert into res_users (id, login, password, active, partner_id, company_id, create_date) VALUES (1, '__system__', NULL, false, 1, 1, now() at time zone 'UTC');
-insert into ir_model_data (name, module, model, noupdate, res_id) VALUES ('user_root', 'base', 'res.users', true, 1);
+insert into ir_model_data (name, module, model, noupdate, res_id, create_date)
+VALUES ('user_root', 'base', 'res.users', true, 1, now() at time zone 'UTC');
 select setval('res_users_id_seq', 1);
 
 insert into res_groups (id, name) VALUES (1, 'Employee');
-insert into ir_model_data (name, module, model, noupdate, res_id) VALUES ('group_user', 'base', 'res.groups', true, 1);
+insert into ir_model_data (name, module, model, noupdate, res_id, create_date)
+VALUES ('group_user', 'base', 'res.groups', true, 1, now() at time zone 'UTC');
```
Comment (rco-odoo) [inline odoo/addons/base/data/base_data.sql:150]: Couldn't we group all those inserts in `ir_model_data`?

File: odoo/modules/db.py (hunk 1 above)
Comment (xmo-odoo) [reply odoo/modules/db.py:68]: > But what about create_uid and write_uid? Those ones are apparently never assigned.

Correct. Didn't really want to touch that part, as they've never been written to these model and there was nothing similar (the weird fields I'm removing only cover the date part of log_access)

File: odoo/addons/base/data/base_data.sql (hunk 2 above)
Comment (xmo-odoo) [reply odoo/addons/base/data/base_data.sql:150]: Could yes, not entirely certain there's a point to it though, base.sql isn't exactly a performance hotspot and the current layout simply pairs ir.model.data entries with the object it's linking to, there's no real reason to change that.

**Comment by xmo-odoo** [comment]
@robodoo r+

**Comment by xmo-odoo** [comment]
@robodoo r-

**Comment by xmo-odoo** [comment]
@robodoo r+

**Comment by xmo-odoo** [comment]
> Linked pull request(s) [odoo/upgrade#1213](https://github.com/odoo/upgrade/pull/1213) not ready. Linked PRs are not staged until all of them are ready.

Just-According-to-Keikaku-001-20180405

File: odoo/addons/base/models/ir_model.py [hunk 3]
```diff
@@ -1990,10 +1988,10 @@ def _update_xmlids(self, data_list, update=False):
     def _build_update_xmlids_query(self, sub_rows, update):
         rowf = "(%s, %s, %s, %s, %s, now() at time zone 'UTC', now() at time zone 'UTC')"
         return """
-            INSERT INTO ir_model_data (module, name, model, res_id, noupdate, date_init, date_update)
+            INSERT INTO ir_model_data (module, name, model, res_id, noupdate, create_date, write_date)
```
Comment (nseinlet) [inline odoo/addons/base/models/ir_model.py:1991]: As now there's a default SQL value, isn't `now() at time zone 'UTC'` useless?

Comment (xmo-odoo) [reply odoo/addons/base/models/ir_model.py:1991]: Yup, well spotted.

**Comment by xmo-odoo** [comment]
@robodoo r+

**Comment by xmo-odoo** [comment]
@robodoo r+ again after rebase ya bastard

**Comment by xmo-odoo** [comment]
@robodoo r+

**Comment by xmo-odoo** [comment]
@robodoo retry

**Comment by xmo-odoo** [comment]
@robodoo r+

**Pull request closed without merging**